timemachine/md/exchange/exchange_mover.py

```diff
@@ -130,7 +139,10 @@ def batch_log_weights_incremental(conf, box, water_idx, new_pos, initial_weights
             # sum interaction energy of all the atoms in a water molecule
             old_water_ixn_nrgs = jnp.sum(self.beta * U_fn_unsummed(conf, box, a_idxs, b_idxs), axis=0)
             old_water_water_ixn_nrgs = jnp.sum(
-                old_water_ixn_nrgs[: (self.num_waters - 1) * 3].reshape((self.num_waters - 1), 3), axis=1
+                old_water_ixn_nrgs[self.starting_water_position :][: (self.num_waters - 1) * 3].reshape(
```

Comment:
Ah on first glance I thought this was a 2D array, but realized it's just doing:
```
old_water_ixn_nrgs[self.starting_water_position : self.starting_water_position + (self.num_waters - 1)]
```

Reply:
Yup, but if you write it that way, JAX complains about dynamic sizes when JITed which is why it is done in two steps.

Reply:
Correct, not sure why it complained. Didn't investigate further.